src/components/Editor/Runners/PythonRunner/PythonRunner.jsx

```diff
@@ -372,14 +376,28 @@ const PythonRunner = () => {
     }
   }
 
+  const singleOutputPanel = outputPanels.length === 1;
+  const showVisualOutput = outputPanels.includes("visual");
+  const showTextOutput = outputPanels.includes("text");
+
   return (
     <div className={`pythonrunner-container`}>
-      {isSplitView ? (
+      {isSplitView || singleOutputPanel ? (
         <>
-          {hasVisualOutput ? (
-            <div className="output-panel output-panel--visual">
+          {hasVisualOutput && showVisualOutput && (
+            <div
+              className={`output-panel output-panel--visual${
```

Comment:
the [classnames](https://www.npmjs.com/package/classnames) package (which we're using elsewhere) offers a nice and easy to read interface for doing this, might be worth taking a look :)

Reply:
Ah yes, okay, I'll take a look - thanks Dan 👍